### PR TITLE
CI: Migrated CI from Travis to GitHub Actions

### LINKED ISSUE #1043: Add CI/CD GitHub Actions and remove other build checks
**Summary:**

The project currently has Codeship, CircleCI, Travis We need to remove these and use Github actions.
Following are the requirements:-

* Generate the apk for every pull request and every commit on the dev and master branch.
* Build checks for every commit on any other branch.

### PR DESCRIPTION
**Pull request by s-ayush2903**
## Issue Fix
Fixes #1043 

## Screenshots
NA

## Description
Added workflow file and removed previous options

##

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by s-ayush2903** [comment]
The artifacts are being generated and uploaded on pushes. But I'm not sure why the CI checks ain't being triggered here as I opened the same PR at my dev with exactly same changes and the CI there was triggered, could be some glitch, maybe another PR would not show the same problem. Anyways, the corresponding push run: [here](https://github.com/s-ayush2903/mobile-wallet/runs/1377235255). The PR of which I was talking: [here](https://github.com/s-ayush2903/mobile-wallet/pull/1/)

**Review by shiv07tiwari — CHANGES_REQUESTED** [review]
@s-ayush2903  Nice work. But there are some things still missing in the file. We need to do a simple build check on every feature branch (all except dev and master) and we need build check + apk for every PR and every commit on dev and master.
You can refer to the following article: https://wkrzywiec.medium.com/github-actions-for-android-first-approach-f616c24aa0f9

**Comment by shiv07tiwari** [comment]
@s-ayush2903  Please add this to the README as well after removing the Travis section.

**Comment by s-ayush2903** [comment]
Fine, I'll definitely make the required changes.

**Comment by s-ayush2903** [comment]
I'm not sure why checks ain't being triggered here at this PR whereas they are working at my PR on my local repo. I'll check if a close and reopen of this PR produces some effect. Or if doesn't then I'll open another PR and then see.

**Comment by shiv07tiwari** [comment]
@s-ayush2903 The actions won't be triggered as the repo is not yet configured for the same. Once we merge your changes, the actions should trigger. Just make the changes using 2 separate yml files for feature branch and master/dev.

**Comment by s-ayush2903** [comment]
Okay fine then. I was wondering that they are being triggered at my fork but not here. Now got it!

File: .github/workflows/feature_branch_ci.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: CI Check for Mobile-Wallet
```
Comment (shiv07tiwari) [inline .github/workflows/feature_branch_ci.yml]: Rename this to Mobile-Wallet Feature Branch CI. It shall refect that it is for the features part.

File: .github/workflows/feature_branch_ci.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+on:
+  push:
+    branches:
+      - '*'
+      - '!dev'
+      - '!master'
+jobs:
+  build:
```
Comment (shiv07tiwari) [inline .github/workflows/feature_branch_ci.yml:11]: Add name alongside runs-on for this jobId

File: .github/workflows/main_branches_ci.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+name: CI Check for Mobile-Wallet
```
Comment (shiv07tiwari) [inline .github/workflows/main_branches_ci.yml]: Same as the last file, rename this. Also IMO we should rename this file to something like master_dev_ci.

File: .github/workflows/pull_request_ci.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+name: CI Check for Mobile-Wallet
```
Comment (shiv07tiwari) [inline .github/workflows/pull_request_ci.yml]: This is exactly the same except for when to trigger the action, right? Just add the pull_request alongside push in the last file. 
on:
  pull_request:
    branches:
      - 'master'
  push:
    branches:
      - 'master'

File: build.gradle [hunk 5]
```diff
@@ -35,6 +35,7 @@ ext {
     compileSdkVersion = 28
     buildToolsVersion = '26.0.2'
     kotlinVersion = "1.3.61"
+    ndkVersion = '21.3.6528147'
```
Comment (shiv07tiwari) [inline build.gradle:38]: Why have you added this line?

File: .github/workflows/main_branches_ci.yml (hunk 3 above)
Comment (s-ayush2903) [reply .github/workflows/main_branches_ci.yml]: Okay, what do you think about the name: `Mobile Wallet CI[Master/Dev]`

File: build.gradle (hunk 5 above)
Comment (s-ayush2903) [reply build.gradle:38]: Actually github actions require ndk version to be specified for the check to run. It is a known issue, not fixing it causes the execution for `stripDebugDebugSymbols` fail which in turn fails entire process. Official Docs: [here](https://developer.android.com/studio/projects/install-ndk#apply-specific-version); known issue: [here](https://github.community/t/android-ndk-on-unix-is-incompatible-with-android-build-tools-3-6-0/17697)

File: .github/workflows/master_dev_ci.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
+name: Mobile-Wallet CI[Master/Dev-Branch]
+on:
+  push:
+    branches:
+      - 'dev'
+      - 'master'
+
+  pull_request:
+    branches:
+      - '*'
```
Comment (shiv07tiwari) [inline .github/workflows/master_dev_ci.yml]: We don't need to add all the branches here. I think adding for dev and master PRs is suffiecient. Just change this and I think we are good to go. :smile:

**Review by shiv07tiwari — CHANGES_REQUESTED** [review]

File: .github/workflows/master_dev_ci.yml (hunk 6 above)
Comment (s-ayush2903) [reply .github/workflows/master_dev_ci.yml]: But what about the hotfix branches then, should we consider build on PRs targeting 'em?

File: .github/workflows/main_branches_ci.yml (hunk 3 above)
Comment (shiv07tiwari) [reply .github/workflows/main_branches_ci.yml]: Yes sure.

File: .github/workflows/master_dev_ci.yml (hunk 6 above)
Comment (shiv07tiwari) [reply .github/workflows/master_dev_ci.yml]: We can add them as per the necessity, but adding for all the branches is not required. I think we can add payment-hub as well here. Add payment-hub in PR and push.

Comment (s-ayush2903) [reply .github/workflows/master_dev_ci.yml]: Fine!

**Review by shiv07tiwari — APPROVED** [review]

**Pull request merged**